pkg/solana/txm/txm.go

```diff
@@ -568,27 +551,48 @@ func (txm *Txm) handleFinalizedSignatureStatus(sig solanaGo.Signature) {
 
 // rebroadcastExpiredTxs attempts to rebroadcast all transactions that are in broadcasted state and have expired.
 // An expired tx is one where it's blockhash lastValidBlockHeight (last valid block number) is smaller than the current block height (block number).
+// The function loops through all expired txes, rebroadcasts them with a new blockhash, and updates the lastValidBlockHeight.
 // If any error occurs during rebroadcast attempt, they are discarded, and the function continues with the next transaction.
 func (txm *Txm) rebroadcastExpiredTxs(ctx context.Context, client client.ReaderWriter) {
 	currBlock, err := client.GetLatestBlock(ctx)
 	if err != nil || currBlock == nil || currBlock.BlockHeight == nil {
 		txm.lggr.Errorw("failed to get current block height", "error", err)
 		return
 	}
-	// Rebroadcast all expired txes using currBlockHeight (current block number)
-	for _, tx := range txm.txs.ListAllExpiredBroadcastedTxs(*currBlock.BlockHeight) {
+
+	// Get all expired broadcasted transactions at current block number. Safe to quit if no txes are found.
+	expiredBroadcastedTxes := txm.txs.ListAllExpiredBroadcastedTxs(*currBlock.BlockHeight)
+	if len(expiredBroadcastedTxes) == 0 {
+		return
+	}
+
+	// Request new blockhash and loop through all expired txes overwriting with new blockhash and rebroadcasting
+	for _, tx := range expiredBroadcastedTxes {
 		txm.lggr.Debugw("transaction expired, rebroadcasting", "id", tx.id, "signature", tx.signatures, "lastValidBlockHeight", tx.lastValidBlockHeight, "currentBlockHeight", *currBlock.BlockHeight)
 		// Removes all signatures associated to tx and cancels context.
 		_, err := txm.txs.Remove(tx.id)
 		if err != nil {
 			txm.lggr.Errorw("failed to remove expired transaction", "id", tx.id, "error", err)
 			continue
 		}
+
+		blockhash, err := client.LatestBlockhash(ctx)
+		if err != nil {
+			txm.lggr.Errorw("failed to get latest blockhash for rebroadcast", "error", err)
+			return
+		}
+		if blockhash == nil || blockhash.Value == nil {
+			txm.lggr.Errorw("nil pointer returned from LatestBlockhash for rebroadcast")
+			return
+		}
```

Comment:
Think it might be better to make this call outside of the for loop? We'll loop through these tx quick enough that I don't think we have to worry about refreshing the blockhash every time.

Reply:
Sure it is. The duplication part of [this definition](https://solana.com/docs/core/transactions#recent-blockhash) made me think that blockhash was unique but after researching a little it only affects two txes that are exactly equal within the same block (very unlikely)
> All transactions include a recent blockhash to act as a timestamp for the transaction. The blockhash is used to prevent duplications and eliminate stale transactions.